internal/biz/usecase/resources/resource_service.go

```diff
@@ -128,6 +128,28 @@ func (uc *Usecase) ReportResource(ctx context.Context, request *v1beta2.ReportRe
 			}
 
 			if err == nil && res != nil {
+				transactionId := request.GetRepresentations().GetMetadata().GetTransactionId()
+
+				// Check if this transaction has already been processed (idempotency)
+				if transactionId != "" && len(res.ReporterResources()) > 0 {
+					reporterResource := res.ReporterResources()[0]
```

Comment:
Once you simplify the signature for the repository method, all of this will not be needed. Only transaction_id parsing will be necessary. 

Small nit: Encapsulate that in a small helper function so you can move it out of update easily, when you do add the transaction_id support. Not end of the world if you don't do it now.